docs/topics/class-methods.rst

```diff
@@ -0,0 +1,29 @@
+===========================
+Class Instantiation Methods
+===========================
+
+These methods create an instance of their implementer class by 
+extracting the components needed for it from the argument that the method takes.
```

Comment:
I think, before we introduce the methods themselves, we need to give some background and provide a list of Scrapy objects that support this.